genesyscloud/external_contacts/genesyscloud_externalcontacts_contact_proxy.go

```diff
@@ -111,37 +111,14 @@ func (p *externalContactsContactsProxy) updateExternalContact(ctx context.Contex
 func getAllExternalContactsFn(ctx context.Context, p *externalContactsContactsProxy) (*[]platformclientv2.Externalcontact, error) {
 	var allExternalContacts []platformclientv2.Externalcontact
 
-	for pageNum := 1; ; pageNum++ {
-		const pageSize = 100
-
-		// The /api/v2/externalcontacts/contacts endpoint can only retrieve 1K records total.
-		// We put a constraint in make sure we never pull then 1,000 records.
-		if pageNum > 10 {
-			fmt.Printf("*******************************************************\n")
-			fmt.Printf("*              Warning                                *\n")
-			fmt.Printf("*******************************************************\n")
-			fmt.Printf("*                                                     *\n")
-			fmt.Printf("* The External Contacts API can only retrieve 1,000   *\n")
-			fmt.Printf("* records. Capping the number of External Contacts    *\n")
-			fmt.Printf("* exported to 1,000.                                  *\n")
-			fmt.Printf("*                                                     *\n")
-			fmt.Printf("*******************************************************\n")
-			return &allExternalContacts, nil
-		}
-
-		externalContacts, _, err := p.externalContactsApi.GetExternalcontactsContacts(pageSize, pageNum, "", "", nil)
-		if err != nil {
-			return nil, fmt.Errorf("Failed to get external contacts: %v", err)
-		}
-
-		if externalContacts.Entities == nil || len(*externalContacts.Entities) == 0 {
-			break
-		}
-
-		for _, externalContact := range *externalContacts.Entities {
-			log.Printf("Dealing with external contact id : %s", *externalContact.Id)
-			allExternalContacts = append(allExternalContacts, externalContact)
-		}
+	externalContacts, _, err := p.externalContactsApi.GetExternalcontactsScanContacts(100, "")
```

Comment:
Would this only be getting the first 100 contacts ?

Reply:
I don't think so. This endpoint is a little different and I believe this means it will retrieve the contacts in batches of 100. I could have misinterpreted so feel free to check. https://developer.genesys.cloud/devapps/api-explorer#get-api-v2-externalcontacts-scan-contacts

Reply:
This is working properly now